### PR TITLE
feat(parental-leave): add explicit union confirmation

### PR DESCRIPTION
**Pull request by miamiultras**
# ...

Attach a link to issue if relevant

Specify what you're trying to achieve

If a union isn't specified, then let the user know that they might lose their rights

Specify why you need to achieve this

Currently its easy for user to accidentally skip that field without being notified, now it has to explicit choose to either use union or not

## Payment
image
## Review
image
## Review edit mode
image

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

**Review by darrikonn — CHANGES_REQUESTED** [review]
The select box should be hidden when no answer (or the answer is no) has been selected.

File: libs/application/templates/parental-leave/src/fields/Review/Review.tsx [hunk 1]
```diff
@@ -320,30 +321,57 @@ export const Review: FC<ReviewScreenProps> = ({
...
+                    label: formatMessage(
+                      parentalLeaveFormMessages.shared.noOptionLabel,
+                    ),
+                    value: NO,
+                  },
+                ]}
+                onSelect={(s: string) => {
+                  setStateful((prev) => ({
+                    ...prev,
+                    useUnion: s as YesOrNo,
```
Comment (darrikonn) [inline libs/application/templates/parental-leave/src/fields/Review/Review.tsx:368]: This probably applies for the other conditional rendering as well (the one blow)... but when no is selected, you've to manually reset the states of relevant inputs. ... so basically, you need to set this as null.

How to replicate?
1. radio = yes
2. select something in select box
3. radio = no
4. continue
5. now, union is set even though the user specified no.

**Review by darrikonn — CHANGES_REQUESTED** [review]

**Review by eirikurn — APPROVED** [review]

File: libs/application/templates/parental-leave/src/fields/Review/Review.tsx (hunk 1 above)
Comment (miamiultras) [reply libs/application/templates/parental-leave/src/fields/Review/Review.tsx:369]: true, done

File: libs/application/ui-shell/src/components/Screen.tsx [hunk 2]
```diff
@@ -255,6 +322,19 @@ const Screen: FC<ScreenProps> = ({
...
+    resetUnionValues()
+  }, [resetUnionValues, watchUseUnion, watchUnion])
+
+  useEffect(() => {
+    resetPensionFundValues()
+  }, [
+    resetPensionFundValues,
+    watchUsePrivatePensionFund,
+    watchPrivatePensionFund,
+    watchPrivatePensionFundPercentage,
```
Comment (darrikonn) [inline libs/application/ui-shell/src/components/Screen.tsx:335]: The night's watch has begun

**Review by darrikonn — APPROVED** [review]

**Review by Thorrik58 — CHANGES_REQUESTED** [review]
I need more info on what we are trying to fix here to give a decent review.
I really dont like the idea of changing the screen component to change something in a specific application.

File: libs/application/ui-shell/src/components/Screen.tsx [hunk 3]
```diff
@@ -40,6 +40,7 @@ import {
   ProblemType,
 } from '@island.is/shared/problem'
 import { handleServerError } from '@island.is/application/ui-components'
+import { NO, YES } from '@island.is/application/templates/parental-leave'
```
[the remark this answers is not in the document]
Comment (miamiultras) [reply libs/application/ui-shell/src/components/Screen.tsx:43]: good point

File: libs/application/ui-shell/src/components/Screen.tsx [hunk 4]
```diff
@@ -239,6 +254,77 @@ const Screen: FC<ScreenProps> = ({
   const { width } = useWindowSize()
   const headerHeight = 85
 
+  const resetUnionValues = useCallback(async () => {
```
[the remark this answers is not in the document]
Comment (miamiultras) [reply libs/application/ui-shell/src/components/Screen.tsx]: Original ticket was to make union more explicit, so now it follows the approach from privatePensionFund, where u have radio button and u have to explicitly choose yes or no.

But Darri found a bug:
image

TLDR: when u click yes, fill in the fields and then click no, u are still submitting fields that shouldn't be there.

I know that this solution is bad workaround, but I tried few others and I can't make it work, because of the app "frameworky" structure and the fact that part of the data is coming from api, part of the data is stored in local state and all of that has to be synced together. If u have better idea I am all ears or let me know if u need more info.

**Review by darrikonn — APPROVED** [review]

**Review by Thorrik58 — APPROVED** [review]
LGTM much better 👍

**Review by qdd1 — APPROVED** [review]
